.github/workflows/release.yaml

```diff
@@ -0,0 +1,36 @@
+name: Release Cloudformation
+on:
+  workflow_dispatch:
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
We should be exclusively using OIDC to assume a role in AWS rather than providing static credentials:

https://github.com/aws-actions/configure-aws-credentials#assuming-a-role

IAM users are a smell. They exist for simplicity and backward compatibility but there's usually a better authentication mechanism. 

And that mechanism is some sort of federation, i.e., verifying an identity the actor already has, often "inherited" or "injected" by the environment somehow, rather than giving it a new set of credentials by which it prove its identity in the target system.

The classic EC2 metadata service/instance profile setup starts to approach this but don't quite reach federation. When you ask the instance metadata API for role credentials you are just reaching an internal-only service that is allow to call STS  for the instance role (associated via instance profile). 

AWS IAM supports federation via SAML and OIDC. SAML is typically used for corporate identity providers (Okta et al). OIDC is a generally better, more modern standard that builds on OAuth. Rather than granting access to a user's resources like typical OAuth, OIDC grants access to an identity.

This allows you to pre-configure AWS IAM to trust identity tokens signed by GitHub. Then when this action obtains an ID token from GitHub and passes it to AWS STS asking for temporary credentials, IAM can evaluates your role trust policy using details from the OIDC claim. This allows you to declare that the role can be assumed only from a certain repository, branch, etc., based on your trust in GitHub to sign those claims into a token accurately/correctly.

https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect

Reply:
> This allows you to pre-configure AWS IAM to trust identity tokens signed by GitHub.

This is implemented on a per-account basis so you probably don't have to do anything to set this up.

Preferably no dependency for CI should be hand-managed. I highly recommend handling this with Terraform:

https://github.com/observeinc/terraform-provider-observe/blob/7ae955da5385240c59835badfbf6107d043f462b/infrastructure/internal-registry.tf

You can tie together the whole configuration and put the role ARN in an Actions variable, no hardcoding anywhere!

Reply:
this is done :+1: